src/components/inputs/rich-text-input/editor.js

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import Button from './button';
+import StyleDropdown from './style-dropdown';
+import MarkButton from './buttons/mark-button';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import useToggleState from '../../../hooks/use-toggle-state';
+import {
+  BoldIcon,
+  ItalicIcon,
+  UnorderedListIcon,
+  OrderedListIcon,
+  UnderlineIcon,
+  RedoIcon,
+} from './icons';
+import { RevertIcon } from '../../icons';
+import Constraints from '../../constraints';
+import FlatButton from '../../buttons/flat-button';
+import { Toolbar, EditorContainer, Container } from './rich-text-input.styles';
+import Divider from './divider';
+import MultiDropdown from './multi-dropdown';
+
+const DEFAULT_NODE = 'paragraph';
+
+const dropdownOptions = [
+  { label: 'Strikethrough', value: 'strikethrough' },
+  { label: 'Superscript', value: 'superscript' },
+  { label: 'Subscript', value: 'subscript' },
+];
+
+const InnerEditor = props => {
```

Comment:
I wanted to use hooks and I couldn't directly use them inside the `Editor` component. This is why it's split like this 😕